### PR TITLE
#806 Joined

### PR DESCRIPTION
**Pull request by VsSekorin**
As per #806

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#853](https://github.com/yegor256/cactoos/issues/853) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#853](https://github.com/yegor256/cactoos/issues/853) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/C63314D6Z/1300929014); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/org/cactoos/iterable/Joined.java [hunk 1]
```diff
@@ -57,6 +58,21 @@ public Joined(final Iterator<Iterable<T>> items) {
...
     }
 
+    /**
+     * Ctor.
+     * @param item First item
+     * @param items Iterable
+     * @since 0.32
+     */
+    public Joined(final T item, final Iterable<T> items) {
+        super(() -> {
```
Comment (Vatavuk) [inline src/main/java/org/cactoos/iterable/Joined.java]: @VsSekorin you can simplify this to:
```java
super(() -> new Joined<>(new IterableOf<>(item), items))
```

File: src/main/java/org/cactoos/list/Joined.java [hunk 2]
```diff
@@ -49,6 +50,18 @@ public Joined(final List<X>... src) {
...
     }
 
+    /**
+     * Ctor.
+     * @param item First item
+     * @param items List
+     * @since 0.32
+     */
+    public Joined(final X item, final List<X> items) {
+        super(() -> Stream.concat(Stream.of(item), items.stream())
```
Comment (Vatavuk) [inline src/main/java/org/cactoos/list/Joined.java]: @VsSekorin you can simplify this to:
```java
super(() -> new Joined<>(new ListOf<>(item), items))
```

File: src/test/java/org/cactoos/iterable/JoinedTest.java [hunk 3]
```diff
@@ -86,4 +86,15 @@ public void joinsIterablesWithSize() {
...
     }
 
+    @Test
+    public void joinItemAndIterable() {
+        MatcherAssert.assertThat(
+            new Joined<>(
+                0,
+                new IterableOf<>(1, 2, 3)
+            ),
+            Matchers.hasItems(0, 1, 2, 3)
```
Comment (Vatavuk) [inline src/test/java/org/cactoos/iterable/JoinedTest.java]: @VsSekorin use `Matchers.contains()`. It is more strict.

File: src/test/java/org/cactoos/list/JoinedTest.java [hunk 4]
```diff
@@ -204,4 +204,15 @@ public void subList() throws Exception {
...
     }
+
+    @Test
+    public void itemAndList() {
+        MatcherAssert.assertThat(
+            new Joined<>(
+                0,
+                new ListOf<>(1, 2, 3)
+            ),
+            Matchers.hasItems(0, 1, 2, 3)
```
Comment (Vatavuk) [inline src/test/java/org/cactoos/list/JoinedTest.java]: @VsSekorin use `Matchers.contains()`.

**Comment by VsSekorin** [comment]
@Vatavuk Thanks. Fixed.

**Comment by Vatavuk** [comment]
@llorllale good to merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#853](https://github.com/yegor256/cactoos/issues/853) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @vatavuk[/z](https://www.0crat.com/u/vatavuk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**